modules/core/src/main/java/org/apache/ignite/internal/processors/cache/verify/IdleVerifyUtility.java

```diff
@@ -85,28 +94,117 @@ public static void checkPartitionsPageCrcSum(
         for (int pageNo = 0; pageNo < pageStore.pages(); pageId++, pageNo++) {
             buf.clear();
 
-            if (cpFlag.get())
-                throw new GridNotIdleException(CLUSTER_NOT_IDLE_MSG);
-
             pageStore.read(pageId, buf, true);
         }
+    }
+
+    /**
+     * Gather updateCounters info.
+     *
+     * @param ign Ignite instance.
+     * @param grpIds Group Id`s.
+     * @return Current group id`s and distribution partitions with update counters per group.
+     */
+    public static T2<Set<Integer>, Map<Integer, Set<Map.Entry<Integer, Long>>>> updCountersSnapshot(
```

Comment:
Also using abbreviations like `updCountersSnapshot` should be avoided in method names, let use self-explanatory `getUpdateCountersSnapshot`